### PR TITLE
fix : queue could be blocked

### PR DESCRIPTION
**Pull request by SkyTNT**
# Description

Surround the code in gradio.queue.Queue.process_event with try...finally... to ensure that the processed events in the queue are cleared to avoid the queue being blocked.

Closes: #2261

# Checklist:

- [X] I have performed a self-review of my own code
- [X] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: gradio/queue.py [hunk 1]
```diff
@@ -220,49 +220,52 @@ async def call_prediction(self, event: Event):
...
+        try:
+            client_awake = await self.gather_event_data(event)
+            if not client_awake:
+                return
+            client_awake = await self.send_message(event, {"msg": "process_starts"})
+            if not client_awake:
+                return
+            begin_time = time.time()
+            response = await self.call_prediction(event)
+            if response.json.get("is_generating", False):
```
[the remark this answers is not in the document]
Comment (SkyTNT) [reply gradio/queue.py:232]: Good idea. I think we also need to log `Response.exception`.

**Comment by freddyaboulton** [comment]
Thank you for the contribution @SkyTNT ! We don't need to raise an exception as the front-end will automatically raise it for us. I added some unit tests too.

**Comment by nttstar** [comment]
Hello, when to update PyPI package to include this PR? It's very critical.

**Comment by abidlabs** [comment]
Hi @nttstar we should have a release out on Monday or Tuesday, but in the meantime you can try the beta release `gradio==3.4b3`

**Comment by liminghao1630** [comment]
Still exist in 3.27.0  @freddyaboulton 
```
2023-04-24 15:39:59,997 - asyncio - ERROR - Task exception was never retrieved
future: <Task finished name='xnedbjj03d_1' coro=<Queue.process_events() done, defined at /Users/drhaogod/miniconda3/envs/daas/lib/python3.10/site-packages/gradio/queueing.py:343> exception=AttributeError("'AsyncRequest' object has no attribute '_json_response_data'")>
Traceback (most recent call last):
  File "/Users/drhaogod/miniconda3/envs/daas/lib/python3.10/site-packages/gradio/queueing.py", line 370, in process_events
    while response.json.get("is_generating", False):
  File "/Users/drhaogod/miniconda3/envs/daas/lib/python3.10/site-packages/gradio/utils.py", line 696, in json
    return self._json_response_data
AttributeError: 'AsyncRequest' object has no attribute '_json_response_data'

python -c "import gradio; print(gradio.__version__)"
3.27.0
```

**Comment by freddyaboulton** [comment]
How can we reproduce @liminghao1630 ?

**Pull request merged**